src/Lean/Util/ServerOptions.lean

```diff
@@ -0,0 +1,87 @@
+import Lean.Util.Paths
+
+namespace Lean
+
+/-- Restriction of `DataValue` that covers exactly those cases that Lean is able to handle when passed via the `-D` flag. -/
+inductive ServerOptionValue where
```

Comment:
Inspired by the discussion above, since these `-D` options also apply on the command, line and to builds it might be worth naming this type `CliOptionValue` or `LeanOptionValue` or `DOptionValue` instead.